metrics/value.go

```diff
@@ -53,9 +84,26 @@ func (v *gaugeVal) Type() string {
 	return "gauge"
 }
 
+func (v *gaugeVal) Keys(yield func(string) bool) {
+	for _, key := range []string{"value"} {
+		if !yield(key) {
+			return
+		}
+	}
+}
+
+func (v *gaugeVal) Value(key string) interface{} {
+	switch key {
+	case "value":
+		return v.Gauge.Value()
+	default:
+		return nil
+	}
+}
+
 func (v *gaugeVal) Values() map[string]interface{} {
 	return map[string]interface{}{
-		"value": v.Value(),
+		"value": v.Gauge.Value(),
```

Comment:
This is a functional change -- is it intentional? Seems correct, but wanted to verify.

Reply:
Related: can we update all the existing `Values` implementations to use the new functions instead? Basically, allocate a map, iterate over all values returned by `keys()` and then assign the value returned by `Value()` (can have a helper function that does this and just call it from all implementations)?

Although I know that `Values()` is now deprecated and unlikely to be used in the future, would like it to share the same underlying source of truth as `Keys()/Value()` combination to avoid any bugs/issues where we update a key or value implementation in one place but not the other.

Reply:
This is equivalent because previously `v.Value()` was calling the embedded metric type's `Value()` method, but since `*gaugeVal` now has a `Value(string)` that no longer works.

I've implemented your suggestion to implement `Values()` based on `Keys()/Value()` in a helper `collectValuesByKey` 👍